components/charts/line-charts/DeveloperHoursLineChart.tsx

```diff
@@ -126,19 +48,68 @@ const legends = [
 ]
 const pointSize = 6
 const useMesh = true
+const yScale = {
+    type: 'linear',
+    min: 0,
+    max: 8
+}
+const colors = ['#3066be', '#da5b38', '#3fc1aa']
 
 const DeveloperHoursLineChart = () => {
+    const { devHours } = useSelector((state: any) => state.developerHours)
+    const { developers } = devHours
+
+    const newDataSource: any = []
```

Comment:
So...dont hate me. but looking at this hours later, all of the logic is GREAT. But, the execution is wrong. We don't want something this heavy being re-run every single render. So let's
1. pull in a `useEffect` that runs on `[developers]`
2. bring in `useState` to store the `dataSource`
3. Run all of the logic in the `useEffect` only if there is no `dataSource` set already